index.bs

```diff
@@ -1350,6 +1352,35 @@ To <dfn for="XRView">obtain the projection matrix</dfn> for a given {{XRView}} |
 </div>
 
 
+Additional Views {#additional-views-support}
+----------
+
+While most devices will either have a single "mono" [=view=] or two "stereo" [=views=], it is possible for devices to have <dfn>additional views</dfn> which do not fall into these categories.
+
+<div class="example">For example, there can be a [=first-person observer view=] used for video capture, or "quad views" where there are two views per eye with differing resolution and fields of view. There can also be CAVE systems which have many views surrounding the user.</div>
+
+While content should be written to assume that there may be any number of views, we expect a significant amount of content to make incorrect assumptions about the {{XRViewerPose/views}} array and thus break when presented with more than two views.
+
+<span class=informative>Because user-agents may have the ability to use mechanisms like reprojection to render to these [=additional views=] in lieu of the content, it is desirable to be able to distinguish between content that plans on handling these [=additional views=] itself and content that is either oblivious to the existence of such [=additional views=] or does not wish to deal with them.</span>
+
+To provide for this, user-agents MAY support an "<dfn>additional-views</dfn>" [=feature descriptor=] as a hint. Content enabling this feature is expected to:
+
+ - Handle any nonzero number of [=views=] in the {{XRViewerPose/views}} array.
+ - Handle the existence of multiple [=views=] that have the same [=view/eye=].
+ - Handle the size of the {{XRViewerPose/views}} array changing from frame to frame. <span class=note>This can happen when video capture is enabled, for example</span>
+
+When "[=additional-views=]" is enabled, the user-agent SHOULD surface any [=additional views=] the device supports to the {{XRSession}}, when necessary.
+
+Note: We recommend content use {{XRSessionInit/optionalFeatures}} to enable "[=additional-views=]" to ensure maximum compatibility.
+
+For devices like CAVE systems where the user-agent is not able to accept stereo views and reproject to simulate the [=additional views=], the user-agent MAY choose surface all [=views=] unconditionally, or it alternatively MAY choose to reject [=immersive sessions=] created without "[=additional-views=]" enabled.
```

Comment:
```suggestion
NOTE: for devices like CAVE systems where the user-agent is not able to accept stereo views and reproject to simulate the [=additional views=], the user-agent MAY choose surface all [=views=] unconditionally, or it alternatively MAY choose to reject [=immersive sessions=] created without "[=additional-views=]" enabled.
```

Reply:
No, this is supposed to be normative, it's explicitly specifying that surfacing 3+ views without the feature enabled is allowed.

Reply:
> No, this is supposed to be normative, it's explicitly specifying that surfacing 3+ views without the feature enabled is allowed.

What is the point of the feature if the UA can ignore it?
It seems strange to allow multiple views while at the same time stating that we don't expect them to work.

Reply:
It's primarily a hint.

In the case of quad views or FPO views it is possible for the UA to reproject a "good enough" approximation. This is what Hololens does when you try recording a session that does not set up an FPO view. So if you write content that is oblivious of extra views, it works. If it is aware of extra views, it can opt in to the better behavior.

However, for CAVE systems, there is no better behavior. Content _must_ be able to handle multiple views, otherwise it will not work. The user agent is given a choice to either reject sessions outright here, or "try it and see if it works". The latter option means that content that doesn't enable this feature still has a chance of working anyway, which is nice to have.

Reply:
hmm, ok. That's a bit confusing.